azurerm/data_source_kubernetes_cluster.go

```diff
@@ -0,0 +1,358 @@
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on AKS Managed Cluster %q (resource group %q): %+v", name, resourceGroup, err)
+	}
+
+	profile, err := kubernetesClustersClient.GetAccessProfiles(ctx, resourceGroup, name, "clusterUser")
+	if err != nil {
+		return fmt.Errorf("Error getting access profile while making Read request on AKS Managed Cluster %q (resource group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if props := resp.ManagedClusterProperties; props != nil {
+		d.Set("dns_prefix", props.DNSPrefix)
+		d.Set("fqdn", props.Fqdn)
+		d.Set("kubernetes_version", props.KubernetesVersion)
+
+		linuxProfile := flattenKubernetesClusterDataSourceLinuxProfile(props.LinuxProfile)
+		if err := d.Set("linux_profile", linuxProfile); err != nil {
+			return fmt.Errorf("Error setting `linux_profile`: %+v", err)
+		}
+
+		agentPoolProfiles := flattenKubernetesClusterDataSourceAgentPoolProfiles(props.AgentPoolProfiles)
+		if err := d.Set("agent_pool_profile", agentPoolProfiles); err != nil {
+			return fmt.Errorf("Error setting `agent_pool_profile`: %+v", err)
+		}
+
+		servicePrincipal := flattenKubernetesClusterDataSourceServicePrincipalProfile(resp.ManagedClusterProperties.ServicePrincipalProfile)
+		if err := d.Set("service_principal", servicePrincipal); err != nil {
+			return fmt.Errorf("Error setting `service_principal`: %+v", err)
+		}
+	}
+
+	kubeConfigRaw, kubeConfig := flattenKubernetesClusterDataSourceAccessProfile(&profile)
+	d.Set("kube_config_raw", kubeConfigRaw)
+
+	if err := d.Set("kube_config", kubeConfig); err != nil {
+		return fmt.Errorf("Error setting `kube_config`: %+v", err)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func flattenKubernetesClusterDataSourceLinuxProfile(input *containerservice.LinuxProfile) []interface{} {
+	values := make(map[string]interface{})
+	sshKeys := make([]interface{}, 0)
+
+	if profile := input; profile != nil {
+		if username := profile.AdminUsername; username != nil {
+			values["admin_username"] = *username
+		}
+
+		if ssh := profile.SSH; ssh != nil {
+			if keys := ssh.PublicKeys; keys != nil {
+				for _, sshKey := range *keys {
+					if keyData := sshKey.KeyData; keyData != nil {
+						outputs := make(map[string]interface{}, 0)
+						outputs["key_data"] = *keyData
+						sshKeys = append(sshKeys, outputs)
+					}
+				}
+			}
+		}
+	}
+
+	values["ssh_key"] = sshKeys
+
+	return []interface{}{values}
+}
+
+func flattenKubernetesClusterDataSourceAgentPoolProfiles(input *[]containerservice.AgentPoolProfile) []interface{} {
+	agentPoolProfiles := make([]interface{}, 0)
+
+	if profiles := input; profiles != nil {
```

Comment:
check could be inverted to reduce nesting.

Reply:
👍 fixed